### ISSUE TITLE
Rule Change: [padding-line-between-statements] Support arrow functions, conditions and member access

### ISSUE THREAD

**Issue by Samuel-Beslogic** [eslint/eslint#15905]
### What rule do you want to change?

padding-line-between-statements

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```json
"padding-line-between-statements": [
  "error",
  {
    "blankLine": "never",
    "prev": "*",
    "next": "*"
  }
],
```
```js
openFeatureRequestPopup = () =>

  this.matDialog.openDialogs.length === 0 &&

  this.matDialog.open(FeatureRequestPopupComponent, { width: '900px' })

    .afterClosed()

    .pipe(takeUntil(this.unsubscribe))

    .subscribe();
```

### What does the rule currently do for this code?

Nothing

### What will the rule do after it's changed?

Allow targetting statements that will catch the extra padding shown above

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @Samuel-Beslogic, thanks for the issue! 

Per our rules policies, [stylistic rules are frozen](https://github.com/eslint/eslint#stylistic-rule-updates), so we are not adding new options to the `padding-line-between-statements` rule. You can always copy the existing rule into a custom rule and make it behave the way you want.
